docs/csharp/asynchronous-programming/async-scenarios.md

```diff
@@ -121,80 +115,83 @@ private async void OnSeeTheDotNetsButtonClick(object sender, RoutedEventArgs e)
+If your program needs to wait on a task, write code that implements the `await` expression in a nonblocking manner. Blocking the current thread as a means to wait for a `Task` item to complete can result in deadlocks and blocked context threads. This programming approach can require more complex error-handling. The following table provides guidance on how to deal with waiting for tasks in a nonblocking way:
 
-  | Use this...          | Instead of this...           | When wishing to do this...                 |
-  |----------------------|------------------------------|--------------------------------------------|
-  | `await`              | `Task.Wait` or `Task.Result` | Retrieving the result of a background task |
-  | `await Task.WhenAny` | `Task.WaitAny`               | Waiting for any task to complete           |
-  | `await Task.WhenAll` | `Task.WaitAll`               | Waiting for all tasks to complete          |
-  | `await Task.Delay`   | `Thread.Sleep`               | Waiting for a period of time               |
+| Task scenario | Current code | Replace with 'await' |
+| --- | --- | --- |
+| _Retrieve the result of a background task_ | `Task.Wait` or `Task.Result` | `await`              |
+| _Wait for any task to complete_            | `Task.WaitAny`               | `await Task.WhenAny` |
+| _Wait for **all** tasks to complete_       | `Task.WaitAll`               | `await Task.WhenAll` |
+| _Wait for some amount of time_             | `Thread.Sleep`               | `await Task.Delay`   |
 
-* **Consider using** `ValueTask` **where possible**
+### Call the ValueTask() method
 
-  Returning a `Task` object from async methods can introduce performance bottlenecks in certain paths. `Task` is a reference type, so using it means allocating an object. In cases where a method declared with the `async` modifier returns a cached result or completes synchronously, the extra allocations can become a significant time cost in performance critical sections of code. It can become costly if those allocations occur in tight loops. For more information, see [generalized async return types](../language-reference/keywords/async.md#return-types).
+When an asynchronous method returns a `Task` object, performance bottlenecks might be introduced in certain paths. Because `Task` is a reference type, an object is allocated when the API is referenced. If a method declared with the `async` modifier returns a cached result or completes synchronously, the extra allocations can accrue significant time costs in performance critical sections of code. This scenario can become costly when the allocations occur in tight loops. For more information, see [generalized async return types](../language-reference/keywords/async.md#return-types).
```

Comment:
word choices:

```suggestion
When an asynchronous method returns a `Task` object, performance bottlenecks might be introduced in certain paths. Because `Task` is a reference type, a `Task` object is allocated from the heap. If a method declared with the `async` modifier returns a cached result or completes synchronously, the extra allocations can accrue significant time costs in performance critical sections of code. This scenario can become costly when the allocations occur in tight loops. For more information, see [generalized async return types](../language-reference/keywords/async.md#return-types).
```